BlueprintUI/Sources/Environment/EnvironmentKey.swift

```diff
@@ -26,4 +26,28 @@ public protocol EnvironmentKey {
     /// The default value that will be vended by an `Environment` for this key if no other value
     /// has been set.
     static var defaultValue: Self.Value { get }
+
+    /// Equivalency check on the `Value`s of `EnvironmentKey`s. This should return false if the
+    /// difference in values affects the measurement or layout of consuming elements.
+    static func isEquivalent(_ lhs: Value, _ rhs: Value) -> Bool
+}
+
+extension EnvironmentKey where Value: Equatable {
+
+    public static func isEquivalent(_ lhs: Value, _ rhs: Value) -> Bool {
+        lhs == rhs
+    }
+}
+
+extension EnvironmentKey {
+    static func areValuesEqual(_ lhs: Any?, _ rhs: Any?) -> Bool {
```

Comment:
Nit/thought:
```suggestion
    @_disfavoredOverload
    static func isEquivalent(_ lhs: Any?, _ rhs: Any?) -> Bool {
```

Reply:
We think it's important to include the word `Value` in here, since the type signature is just two `Any`s, which could be construed to be the keys. Perhaps `areValuesEquivalent`?